tmux/tmux
# Add options and arguments for display-menu to set border, border style, and style

~~:warning: The merge base of this PR should be [`afh:afh-popup-style-arguments`](https://github.com/afh/tmux/tree/afh-popup-style-arguments) from #2931 instead of [`tmux:master`](https://github.com/tmux/tmux).
:information_source: To see only the relevant changes please compare [`afh:afh-popup-style-arguments` with`afh:afh-menu-style-opts-args`](https://github.com/afh/tmux/compare/afh-popup-style-arguments...afh:afh-menu-style-opts-args?expand=1).~~ Update: #2931 has been applied, so master as a merge base for this PR is now correct 🎉 

This PR brings the same styling capabilites from popup to menu.

To test:
* Build and run tmux `./tmux -vv -f/dev/null -Ltest`
* Set menu-border-lines, menu-style options and display menu (see `test_menu_style.sh` below)
* Use `-B`, `-b`, `-s, `-S` arguments when displaying a menu (see `test_menu_style.sh` below)

<details><summary><code>test_menu_style.sh</code></summary>

```
#!/bin/ksh

./tmux set -gwU menu-border-lines
./tmux set -gw menu-border-lines double
./tmux set -gw menu-border-style fg=black,bg=yellow
./tmux set -gw menu-style fg=magenta,bg=cyan
./tmux set -gw mode-style fg=yellow,bg=black
./tmux menu -T 'Styled via Options' '#[fg=red,bg=blue]foo' '' '' -bar '' '' '' baz '' ''

./tmux menu -B -T 'Styled via Options' '#[fg=red,bg=blue]foo' '' '' -bar '' '' '' baz '' ''

./tmux menu -b heavy -S fg=blue,bg=red -s fg=magenta,bg=cyan -T 'Styled via Arguments' foo '' '' -bar '' '' '' baz '' ''

./tmux menu -B -b heavy -S fg=blue,bg=red -s fg=magenta,bg=cyan -T 'Styled without Border' foo '' '' -bar '' '' '' baz '' ''
```

</details>

| | |
| --- | --- |
| Styled with Options | ![2021-10-23-101801_905x562_scrot](https://user-images.githubusercontent.com/16507/138549065-5711b50b-eb22-47e0-aa36-3f49dd754333.png)|
| Styled without Border | ![2021-10-23-103011_905x562_scrot](https://user-images.githubusercontent.com/16507/138549185-b0bb8e3e-d409-4861-9be9-2a6bae636ef0.png)|
| Styled with Arguments | ![2021-10-23-101809_905x562_scrot](https://user-images.githubusercontent.com/16507/138549066-c09ffb31-c528-40c9-b4ac-886414c7d8a3.png)|
| Styled without Border | ![2021-10-23-101815_905x562_scrot](https://user-images.githubusercontent.com/16507/138549068-09f90e58-690e-4782-af9f-5667955ec091.png)|
